### PR TITLE
Add basic example and update README

### PR DESCRIPTION
**Pull request by nrempel**
This pull request adds more content to the README and adds a basic example.

### CODE REVIEW

File: instant-distance/examples/colors.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    let values = vec!["red", "green", "blue"];
+
+    let map = Builder::default().build(points, values);
+    let mut search = Search::default();
+
+    let cambridge_blue = Point(163, 193, 173);
+
+    let closest_point = map.search(&cambridge_blue, &mut search).next().unwrap();
+
+    println!("{:?}", closest_point.value);
```
Comment (nrempel) [inline instant-distance/examples/colors.rs:15]: @djc is this the intended way to retrieve the value? For me, the value changes every time. However, the pointID remains constant. To me it looked like HnswMap.values is sorted prior to being exposed here, but maybe I'm missing something: https://github.com/nrempel/instant-distance/blob/4a4ee8890717db205a38683a11b0c89559d5d480/instant-distance/src/lib.rs#L163-L163

File: README.md [hunk 2]
```diff
@@ -6,10 +6,71 @@
...
+instant-segment = "0.5.0"
+```
+
+## Example
+
+```rust
+use instant_distance::{Builder, Search};
+
+fn main() {
+    //
```
Comment (djc) [inline README.md:35]: Probably should just remove this?

**Review by djc — COMMENTED** [review]
Nice!

File: instant-distance/examples/colors.rs (hunk 1 above)
Comment (djc) [reply instant-distance/examples/colors.rs:15]: Yeah, that should be the way it works. Are you saying this does not print "blue" every time?

Comment (nrempel) [reply instant-distance/examples/colors.rs:15]: Correct:
```
nbrempel@Nick-IDS ~/P/instant-distance (main)> cargo run --example colors                    (base) 
   Compiling instant-distance v0.5.0 (/Users/nbrempel/Projects/instant-distance/instant-distance)
    Finished dev [unoptimized + debuginfo] target(s) in 5.92s
     Running `target/debug/examples/colors`
"red"
nbrempel@Nick-IDS ~/P/instant-distance (nrempel/update-readme)> cargo run --example colors   (base) 
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/examples/colors`
"red"
nbrempel@Nick-IDS ~/P/instant-distance (nrempel/update-readme)> cargo run --example colors   (base) 
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/examples/colors`
"red"
nbrempel@Nick-IDS ~/P/instant-distance (nrempel/update-readme)> cargo run --example colors   (base) 
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/examples/colors`
"red"
nbrempel@Nick-IDS ~/P/instant-distance (nrempel/update-readme)> cargo run --example colors   (base) 
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/examples/colors`
"blue"
nbrempel@Nick-IDS ~/P/instant-distance (nrempel/update-readme)> cargo run --example colors   (base) 
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
     Running `target/debug/examples/colors`
"red"
nbrempel@Nick-IDS ~/P/instant-distance (nrempel/update-readme)> cargo run --example colors   (base) 
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
     Running `target/debug/examples/colors`
"red"
nbrempel@Nick-IDS ~/P/instant-distance (nrempel/update-readme)> cargo run --example colors   (base) 
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
     Running `target/debug/examples/colors`
"red"
nbrempel@Nick-IDS ~/P/instant-distance (nrempel/update-readme)> cargo run --example colors   (base) 
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/examples/colors`
"red"
nbrempel@Nick-IDS ~/P/instant-distance (nrempel/update-readme)> cargo run --example colors   (base) 
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/examples/colors`
"green"

```

File: README.md (hunk 2 above)
Comment (nrempel) [reply README.md:35]: It is redundant, but I do also like having an example right in the readme for clarity.

File: README.md [hunk 3]
```diff
@@ -6,10 +6,71 @@
...
+    let points = vec![Point(255, 0, 0), Point(255, 0, 0), Point(255, 0, 0)];
```
Comment (djc) [inline README.md:35]: I think we have a PEBKAC here! It might also be misleading that "cambridge blue" is actually closer to green than to blue?

Comment (nrempel) [reply README.md:35]: oh jeeze, that will do it 🤦🏼‍♂️

**Comment by nrempel** [comment]
I fixed that bug in the example and changed the example color to something less confusing.

**Comment by djc** [comment]
Nice, thanks!

**Pull request merged**